---
title: "Introduction to motiftools"
author: "Diego Diez"
date: "`r BiocStyle::doc_date()`"
package: "`r BiocStyle::pkg_ver('motiftools')`"
output:
    rmarkdown::html_document
abstract: >
  Introduction to the motiftools package.
vignette: >
 %\VignetteIndexEntry{Introduction to motiftools}
 %\VignetteEngine{knitr::rmarkdown}
 %\VignetteEncoding{UTF-8}
---


```{r setup, include=FALSE}
library(BiocStyle)
library(ggplot2)
theme_set(theme_minimal())
```

# Introduction

motiftools is a package for the analysis and interpretation of sequence motifs. Its initial implementation is focused on protein motif analysis obtained from the MEME suite, but its functionality may extend to other applications and tools, including analysis of nucleotide sequences and motif analysis tools found in Bioconductor or CRAN.

Motif ranges are stored as `RangedData` objects (`r Biocpkg("IRanges")`) package). This allows to include metadata, like p-value, score and other output from the search process.

```{r, message=FALSE}
library(motiftools)
library(motifStack)
```

# Plotting functions

## Alignments

You can align any two sequences of letters. If the argument `score.matrix` is missing then a empirical scoring matrix will be generated from the sequences themeselves.

```{r}
align("FOO", "GOO", type = "global")

align("AHSSFOO", "GOODFDKD", type = "global")$alignment
```

For protein sequences you can specify a particular substitution matrix, like for example BLOSUM62.

```{r}
data("BLOSUM62", package = "Biostrings")
align("EGGALAPA", "DGGIVLPSAV", score.matrix = BLOSUM62, type = "global")$alignment
align("EGGALAPA", "DGGIVFFRF", score.matrix = BLOSUM62, type = "local")$alignment
```

## Motifs

```{r}
# load MEME results.
f <- system.file("files/meme_ras/meme.xml", package = "motiftools")
x_meme <- readMEME(f)
m_meme <- getMotifMatrix(x_meme)
plotMatrix(m_meme, fill = c("white", "grey"))

# load FIMO results.
f <- system.file("files/fimo_ras/fimo.xml", package = "motiftools")
x_fimo <- readFIMO(f)
m_fimo <- getMotifMatrix(x_fimo)
plotMatrix(list(m_meme, m_fimo), fill = c("white", "grey"))

# load MAST results.
f <- system.file("files/mast_ras/mast.xml", package = "motiftools")
x_mast <- readMAST(f)
m_mast <- getMotifMatrix(x_mast)
plotMatrix(list(m_meme, m_fimo, m_mast), fill = c("white", "grey"))

# highlight some sequences.
h <- grepl("RASH|RASK", sequenceNames(x_meme))
names(h) <- sequenceNames(x_meme)
plotMatrix(list(m_meme, m_fimo, m_mast), fill = c("white", "grey"), high = h, high.col = c("black", "orange"))

# plot a motif model logo (from MEME output).
p <- pwm(x_meme)[["motif_4"]]
plotMotifLogo(p, col = rainbow(20))
```

## Sequences

```{r}
m <- matrix(sample(1:5, 100*10, replace = TRUE), ncol = 100, nrow = 10, dimnames = list(letters[1:10], 1:100))
h <- hclust(dist(m))
plotConservationMatrix(m, h)
```

