<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Introduction to motiftools &bull; motiftools</title><!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous"><script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous"><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../jquery.sticky-kit.min.js"></script><script src="../pkgdown.js"></script><!-- mathjax --><script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <div class="container template-vignette">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">motiftools</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../articles/index.html">Articles</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9">
    <div class="page-header toc-ignore">
      <h1>Introduction to motiftools</h1>
                        <h4 class="author">Diego Diez</h4>
            
            <h4 class="date">8 November 2016</h4>
          </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Introduction to the motiftools package.</p>
    </div>
    
<div class="contents">
<div id="introduction" class="section level1">
<h1 class="hasAnchor"><html><body><a href="#introduction" class="anchor"> </a></body></html>Introduction</h1>
<p>motiftools is a package for the analysis and interpretation of sequence motifs. Its initial implementation is focused on protein motif analysis obtained from the MEME suite, but its functionality may extend to other applications and tools, including analysis of nucleotide sequences and motif analysis tools found in Bioconductor or CRAN.</p>
<p>Motif ranges are stored as <code>RangedData</code> objects (<em><a href="http://bioconductor.org/packages/IRanges">IRanges</a></em>) package). This allows to include metadata, like p-value, score and other output from the search process.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(motiftools)
<span class="kw">library</span>(motifStack)
<span class="kw">library</span>(Biostrings)
<span class="kw">library</span>(ggtree)</code></pre></div>
</div>
<div id="plotting-functions" class="section level1">
<h1 class="hasAnchor"><html><body><a href="#plotting-functions" class="anchor"> </a></body></html>Plotting functions</h1>
<div id="alignments" class="section level2">
<h2 class="hasAnchor"><html><body><a href="#alignments" class="anchor"> </a></body></html>Alignments</h2>
<p>You can align two protein sequences specifying a particular substitution matrix (e.g.&nbsp;BLOSUM62).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">"BLOSUM62"</span>, <span class="dt">package =</span> <span class="st">"Biostrings"</span>)
<span class="kw"><a href="../reference/align.html">align</a></span>(<span class="st">"EGGALAPA"</span>, <span class="st">"DGGIVLPSAV"</span>, <span class="dt">score.matrix =</span> BLOSUM62, <span class="dt">type =</span> <span class="st">"global"</span>)$alignment</code></pre></div>
<pre><code>##   1 2 3 4 5 6 7 8 9 10
## x E G G - A L A P -  A
## y D G G I V L - P S  A</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/align.html">align</a></span>(<span class="st">"EGGALAPA"</span>, <span class="st">"DGGIVFFRF"</span>, <span class="dt">score.matrix =</span> BLOSUM62, <span class="dt">type =</span> <span class="st">"local"</span>)$alignment</code></pre></div>
<pre><code>##   1 2 3 4 5 6
## x E G G A L A
## y D G G - I V</code></pre>
</div>
<div id="motifs" class="section level2">
<h2 class="hasAnchor"><html><body><a href="#motifs" class="anchor"> </a></body></html>Motifs</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># load MEME results.</span>
f &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"files/meme_ras/meme.xml"</span>, <span class="dt">package =</span> <span class="st">"motiftools"</span>)
x_meme &lt;-<span class="st"> </span><span class="kw"><a href="../reference/readMEME.html">readMEME</a></span>(f)
m_meme &lt;-<span class="st"> </span><span class="kw"><a href="../reference/getMotifMatrix.html">getMotifMatrix</a></span>(x_meme)
<span class="kw"><a href="../reference/plotMotifMatrix.html">plotMotifMatrix</a></span>(m_meme, <span class="dt">fill =</span> <span class="kw">c</span>(<span class="st">"white"</span>, <span class="st">"grey"</span>))</code></pre></div>
<p><img src="introduction_files/figure-html/unnamed-chunk-12-1.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># load FIMO results.</span>
f &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"files/fimo_ras/fimo.xml"</span>, <span class="dt">package =</span> <span class="st">"motiftools"</span>)
x_fimo &lt;-<span class="st"> </span><span class="kw"><a href="../reference/readFIMO.html">readFIMO</a></span>(f)
m_fimo &lt;-<span class="st"> </span><span class="kw"><a href="../reference/getMotifMatrix.html">getMotifMatrix</a></span>(x_fimo)
<span class="kw"><a href="../reference/plotMotifMatrix.html">plotMotifMatrix</a></span>(<span class="kw">list</span>(<span class="dt">meme =</span> m_meme, <span class="dt">fimo =</span> m_fimo), <span class="dt">fill =</span> <span class="kw">c</span>(<span class="st">"white"</span>, <span class="st">"grey"</span>), <span class="dt">col =</span> <span class="st">"black"</span>)</code></pre></div>
<p><img src="introduction_files/figure-html/unnamed-chunk-12-2.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># load MAST results.</span>
f &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"files/mast_ras/mast.xml"</span>, <span class="dt">package =</span> <span class="st">"motiftools"</span>)
x_mast &lt;-<span class="st"> </span><span class="kw"><a href="../reference/readMAST.html">readMAST</a></span>(f)
m_mast &lt;-<span class="st"> </span><span class="kw"><a href="../reference/getMotifMatrix.html">getMotifMatrix</a></span>(x_mast)
<span class="kw"><a href="../reference/plotMotifMatrix.html">plotMotifMatrix</a></span>(<span class="kw">list</span>(<span class="st">"MEME motifs"</span> =<span class="st"> </span>m_meme, <span class="st">"FIMO motifs"</span> =<span class="st"> </span>m_fimo, <span class="st">"MAST motifs"</span> =<span class="st"> </span>m_mast), <span class="dt">fill =</span> <span class="kw">c</span>(<span class="st">"white"</span>, <span class="st">"grey"</span>), <span class="dt">col =</span> <span class="st">"black"</span>)</code></pre></div>
<p><img src="introduction_files/figure-html/unnamed-chunk-12-3.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># highlight some sequences.</span>
h &lt;-<span class="st"> </span><span class="kw">grepl</span>(<span class="st">"RASH|RASK"</span>, <span class="kw"><a href="../reference/MotifSearchResult-class.html">sequenceNames</a></span>(x_meme))
<span class="kw">names</span>(h) &lt;-<span class="st"> </span><span class="kw"><a href="../reference/MotifSearchResult-class.html">sequenceNames</a></span>(x_meme)
<span class="kw"><a href="../reference/plotMotifMatrix.html">plotMotifMatrix</a></span>(<span class="kw">list</span>(<span class="st">"MEME motifs"</span> =<span class="st"> </span>m_meme, <span class="st">"FIMO motifs"</span> =<span class="st"> </span>m_fimo, <span class="st">"MAST motifs"</span> =<span class="st"> </span>m_mast), <span class="dt">fill =</span> <span class="kw">c</span>(<span class="st">"white"</span>, <span class="st">"grey"</span>), <span class="dt">col =</span> <span class="st">"black"</span>, <span class="dt">high =</span> h, <span class="dt">high.col =</span> <span class="kw">c</span>(<span class="st">"black"</span>, <span class="st">"orange"</span>))</code></pre></div>
<p><img src="introduction_files/figure-html/unnamed-chunk-12-4.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># plot a motif model logo (from MEME output).</span>
p &lt;-<span class="st"> </span><span class="kw"><a href="../reference/MotifSearchResult-class.html">pwm</a></span>(x_meme)[[<span class="st">"motif_4"</span>]]
<span class="kw">plotMotifLogo</span>(p, <span class="dt">col =</span> <span class="kw">rainbow</span>(<span class="dv">20</span>))</code></pre></div>
<pre><code>## Warning in plotMotifLogo(p, col = rainbow(20)): background length is
## shorter than number of rows. Will use default background</code></pre>
<p><img src="introduction_files/figure-html/unnamed-chunk-12-5.png" width="672"></p>
<p>You can compare the motifs obtained with MEME using the TOMTOM tool. Then you can read the results and look at the comparison matrix.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">f &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">"files/tomtom_ras/tomtom.xml"</span>, <span class="dt">package =</span> <span class="st">"motiftools"</span>)
x_tomtom &lt;-<span class="st"> </span><span class="kw"><a href="../reference/readTOMTOM.html">readTOMTOM</a></span>(f)
<span class="kw"><a href="../reference/plotMotifMatches.html">plotMotifMatches</a></span>(x_tomtom)</code></pre></div>
<p><img src="introduction_files/figure-html/unnamed-chunk-13-1.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plotMotifMatches.html">plotMotifMatches</a></span>(x_tomtom, <span class="dt">fill =</span> <span class="st">"e_value"</span>)</code></pre></div>
<p><img src="introduction_files/figure-html/unnamed-chunk-13-2.png" width="672"></p>
</div>
<div id="sequences" class="section level2">
<h2 class="hasAnchor"><html><body><a href="#sequences" class="anchor"> </a></body></html>Sequences</h2>
<p>We can visualize MSA conservation together with a phylogenetic tree.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ras_tree &lt;-<span class="st"> </span><span class="kw">read.tree</span>(<span class="kw">system.file</span>(<span class="st">"files/ras.tree"</span>, <span class="dt">package =</span> <span class="st">"motiftools"</span>))
ras_aln &lt;-<span class="st"> </span><span class="kw">readAAMultipleAlignment</span>(<span class="kw">system.file</span>(<span class="st">"files/ras.faln"</span>, <span class="dt">package =</span> <span class="st">"motiftools"</span>))
<span class="kw">rownames</span>(ras_aln) &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="st">" .*"</span>, <span class="st">""</span>, <span class="kw">rownames</span>(ras_aln)) <span class="co"># need to fix sequence names.</span>
<span class="kw"><a href="../reference/plotConservationMatrix.html">plotConservationMatrix</a></span>(<span class="kw"><a href="../reference/conservationMatrix.html">conservationMatrix</a></span>(ras_aln), ras_tree, <span class="dt">show.tips =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="introduction_files/figure-html/unnamed-chunk-14-1.png" width="672"></p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2>Contents</h2>
      <ul class="nav nav-pills nav-stacked"><li><a href="#introduction">Introduction</a></li>
      <li><a href="#plotting-functions">Plotting functions</a><ul class="nav nav-pills nav-stacked"><li><a href="#alignments">Alignments</a></li>
      <li><a href="#motifs">Motifs</a></li>
      <li><a href="#sequences">Sequences</a></li>
      </ul></li>
      </ul></div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Diego Diez.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://hadley.github.io/pkgdown/">pkgdown</a>.</p>
</div>

      </footer></div>

  </body></html>
